### DISCUSSION TITLE
Test Plans  for the Runner

### DISCUSSION THREAD

**Discussion by Frankenleg** [usebruno/bruno#2292] [Ideas]
Goal:  
Run only specific API calls during specific testing scenarios.

Need:
I need to run regression testing in a production API release. I know this is not ideal, but my reality is  we don't own some of the other microservices deployed in tandem with our APIs.   When doing the regression testing, I don't want to send certain post/put requests to update production data.  I think this could also be useful for other specific use cases as well. Production/Quality environments,  new feature development, etc.   I currently have to re-organize these requests into specific folders, which is not ideal for organization.

Ideas:
- The http request defines an array of valid test plans.
- The runner accepts an optional 'testplan' array pram to filter tests.
- The runner tab/page lists a distinct multi-list selection of testplans found in the collection
- The runner gets a distinct list of requests matching the search criteria, ordered by the seq.

Possible Examples:
```
meta {
  name: Example Request
  type: http
  seq: 2
  testplan:  [production, quality, my new feature]
}
```
OR
```
meta {
  name: Example Request
  type: http
  seq: 2
}

get {
  url: 
  body: none
  auth: none
}

assert {
  res.status: eq 200
  res.body: neq 
}

testplan: {[
  production, 
  development,
  my new feature 
]
}
```

**Comment by nbran** [comment]
+1, missing key feature.

**Comment by davidsebag0705** [comment]
Someone suggests the idea to have tags for requests (in meta for example, see https://github.com/usebruno/bruno/issues/1039) and have the ability to launch the runner with params "include theses tags" and/or "exclude these tags". Is it the same kind of idea as yours ?
